Add functionality to automatically retrieve/generate access token

cemulate opened the pull request:
You can now initialize an apple reporter with an access token (if you have one) **or** the account password.
When you use the latter option, you can call `retrieveAccessToken()` or `retrieveAccessToken({ generateNewIfNeeded: true })` to configure the access token before using the rest of the library.

cemulate committed:
Add functionality to automatically retrieve/generate access token

cemulate committed:
Skip retrieving the access token if it's already been done, unless the user asks to retrieve it anyway.

cemulate committed:
If an access token is given instead of a password, don't throw an error in retrieveAccessToken() unless we actually need to get the token (i.e. forceRetrieve is set)

cemulate committed:
Bump semver

aaronbean commented:
Any idea why the build is failing?  Would look better if it didn't.  Perhaps the tests are not working well for the changes?

cemulate committed:
This syntax works for specifiying the tests to Mocha?

cemulate committed:
Fix failing test

Since we were building the json body in a slightly different order
(password or access token gets put on last), this was failing because it
assumed a certain order.

cemulate committed:
Add some basic tests for the retrieveAccessToken() functionality

cemulate committed:
Fix pre-existing lint failures. Now the output is 100% green!!

aaronbean reviewed (approved):
excellent job eliminating underscores!
